test/src/test/kotlin/graphql/nadel/tests/hooks/batching-of-hydration-list-with-partition.kt

```diff
@@ -0,0 +1,36 @@
+package graphql.nadel.tests.hooks
+
+import graphql.nadel.Nadel
+import graphql.nadel.enginekt.NadelEngineExecutionHooks
+import graphql.nadel.enginekt.blueprint.NadelBatchHydrationFieldInstruction
+import graphql.nadel.enginekt.blueprint.NadelGenericHydrationInstruction
+import graphql.nadel.enginekt.transform.artificial.NadelAliasHelper
+import graphql.nadel.enginekt.transform.result.json.JsonNode
+import graphql.nadel.tests.EngineTestHook
+import graphql.nadel.tests.NadelEngineType
+import graphql.nadel.tests.UseHook
+
+private class BatchHydrationHooks : NadelEngineExecutionHooks {
+    override fun <T : NadelGenericHydrationInstruction> getHydrationInstruction(
+        instructions: List<T>,
+        parentNode: JsonNode,
+        aliasHelper: NadelAliasHelper
+    ): T {
+        return instructions[0]
+    }
+
+    override fun <T> partitionArgumentList(
+        args: List<T>,
+        instruction: NadelBatchHydrationFieldInstruction
+    ): List<List<T>> {
+        val partition = args.partition { (it as String).endsWith("2") || it.endsWith("4") }
```

Comment:
Can we make a closer to real example for the AGG?

e.g. do `(it as String).substringBefore("/")` and have the values be like `cloud-1/fish-1` and `cloud-1/human-1` etc?

I'm kind of struggling to see how this does multi tenanted ness, I assume because we split each query to only be `cloud-1` and `cloud-2` and the AGG does the rest when transforming the query??

Reply:
done!

Reply:
> I'm kind of struggling to see how this does multi tenanted ness

So we will split arguments in separate batches by cloudIds (similar how we do it for `batchSize`) and AGG then executes these batches separately. We will only do it for sharded services. So before executing the hydration request, AGG will lookup the shard url based on the argument values in the batch.

Reply:
> separate batches by cloudIds
we can actually split by shard url since different cloud is might live in the same shard